### PR TITLE
Add Cookie Cutter to Safari Extension

### PR DESCRIPTION
**Pull request by evancrow**
Adds Cookie Cutter to the mobile Safari Extension.

Things left to do:
- [x] Add user preferences
- [x] User education
- [x] Run yarn for NeevaForSafariExtension with bootstrap

Note that there is no feature flag here, the extension doesn't support that. Will wait on merging till we know this is what we want to do.

cc @jlynch630 @hwiatneeva 

### How was this tested?
- [x] I tested this manually

### Screenshots and screen recordings

https://user-images.githubusercontent.com/37356984/176954489-0c50c7f0-fa4e-49b3-b620-bae454eb465e.mp4

## Issues addressed
Fixed: #3665

### CODE REVIEW

**Comment by evancrow** [comment]
Marking as ready for code review, can discuss in Magpie weekly.

File: NeevaForSafari/NeevaForSafariExtension/Resources/cookieCutterContentScript.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Inject the Cookie Cutter script into the webpage.
+const scriptUrl = browser.runtime.getURL('scripts/cookieCutterEngine.js');
+const element = document.createElement('script');
+element.src = scriptUrl;
+document.head.appendChild(element);
+
+// Add a listner to pass messages from the Cookie Cutter script
```
Comment (jlynch630) [inline NeevaForSafari/NeevaForSafariExtension/Resources/cookieCutterContentScript.js:7]: nit: listener

**Review by jlynch630 — APPROVED** [review]
LGTM, just a small nit

**Comment by hwiatneeva** [comment]
- from design: LGTM as-is
- granular breakdowns of types of non essential cookies:  scope as P2 nice to have

**Review by darinwf — APPROVED** [review]
I mostly focused my review on the native code, deferring to John for the web code. LGTM

**Pull request merged**